hypershift-operator/controllers/nodepool/openstack/openstack.go

```diff
@@ -72,3 +76,62 @@ func MachineTemplateSpec(hcluster *hyperv1.HostedCluster, nodePool *hyperv1.Node
 	}
 	return openStackMachineTemplate, nil
 }
+
+func ReconcileOpenStackImageCR(ctx context.Context, client client.Client, createOrUpdate upsert.CreateOrUpdateFN, hcluster *hyperv1.HostedCluster, controlPlaneNamespace string, release *releaseinfo.ReleaseImage) error {
+	openStackImage := orc.Image{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "rhcos-" + hcluster.Name,
+			Namespace: controlPlaneNamespace,
+			// TODO: add proper cleanup in CAPI resources cleanup
+			OwnerReferences: []metav1.OwnerReference{
+				{
+					APIVersion: hcluster.APIVersion,
+					Kind:       hcluster.Kind,
+					Name:       hcluster.Name,
+					UID:        hcluster.UID,
+				},
+			},
+		},
```

Comment:
I'm pretty sure this is the problem: owner references can't be cross-namespace. Because this reference is invalid, the k8s gc will automatically delete it at some point. When that happens this controller will presumably re-create it and we get into the weird situation we see in the logs where we're referencing a deleted object.

The object needs to be owned by something in the same namespace. The HostedCP is the obvious candidate.

Reply:
This was a mistake, I'm not sure why I used that namespace :/